python/ray/data/_internal/table_block.py

```diff
@@ -119,6 +120,10 @@ def will_build_yield_copy(self) -> bool:
 
     def build(self) -> Block:
         if self._columns:
+            self._columns = {
+                key: convert_udf_returns_to_numpy(col)
+                for key, col in self._columns.items()
+            }
```

Comment:
@ericl #35272 doesn't cover the following case. This change is to fix this issue. Lmk if you think there is a better fix.

```python
import ray

ds = ray.data.read_images("s3://anonymous@air-example-data/AnimalDetection/JPEGImages")
print(ds.schema())

import numpy as np
import torch
from torchvision import transforms
from torchvision.models.detection import (FasterRCNN_ResNet50_FPN_V2_Weights,
                                          fasterrcnn_resnet50_fpn_v2)
from typing import Dict

def preprocess_image(data: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
    weights = FasterRCNN_ResNet50_FPN_V2_Weights.DEFAULT
    preprocessor = transforms.Compose(
        [transforms.ToTensor(), weights.transforms()]
    )
    return {
        "image": data["image"],
        "transformed": preprocessor(data["image"]),
    }

ds = ds.map(preprocess_image)

print(ds.schema())
```

Reply:
cc @ericl for this change

Reply:
Hmm I feel like this should be covered by https://github.com/ray-project/ray/pull/35272/files#diff-a637aa9db5408fe440f8b335a480e66e6636d97e71dc4bad164c356ac43deafdR180

Do we know why it's not going down that codepath?

Reply:
That one seems to only cover map_batches, not map.

Reply:
I think this is fixed in master (see test_numpy_support.py::test_scalar_array_like).